homeassistant/components/combined_energy/sensor.py

```diff
@@ -0,0 +1,529 @@
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+            device_class=SensorDeviceClass.ENERGY,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="power_consumption_battery",
+            name="Power Export Battery",
+            icon="mdi:home-battery",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfPower.KILO_WATT,
+            device_class=SensorDeviceClass.POWER,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="power_factor_a",
+            name="Power Factor A",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="%",
+            device_class=SensorDeviceClass.POWER_FACTOR,
+        ),
+        SensorEntityDescription(
+            key="power_factor_b",
+            name="Power Factor B",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="%",
+            device_class=SensorDeviceClass.POWER_FACTOR,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="power_factor_c",
+            name="Power Factor C",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="%",
+            device_class=SensorDeviceClass.POWER_FACTOR,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="voltage_a",
+            name="Voltage A",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+        ),
+        SensorEntityDescription(
+            key="voltage_b",
+            name="Voltage B",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="voltage_c",
+            name="Voltage C",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            entity_registry_enabled_default=False,
+        ),
+    ],
+    "GENERIC_CONSUMER": SENSOR_DESCRIPTIONS_GENERIC_CONSUMER,
+    "ENERGY_BALANCE": SENSOR_DESCRIPTIONS_GENERIC_CONSUMER,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+
+    api: CombinedEnergy = hass.data[DOMAIN][entry.entry_id][DATA_API_CLIENT]
+    installation: Installation = hass.data[DOMAIN][entry.entry_id][DATA_INSTALLATION]
+
+    # Initialise services
+    log_session = CombinedEnergyLogSessionService(hass, api)
```

Comment:
Please move the creation of the coordinator to `__init__.py` and call `async_config_entry_first_refresh`